python/rmm/rmm/_lib/device_buffer.pyx

```diff
@@ -345,6 +365,14 @@ cdef class DeviceBuffer:
     cdef void* c_data(self) except *:
         return self.c_obj.get()[0].data()
 
+    cdef void c_prefetch(self,
```

Comment:
Having a separate function for this seems excessive when it could just be inlined into `prefetch` above. To keep things concise you can set the first argument with a ternary in Python and Cython shouldn't have a problem with it; `prefetch(cuda_device_id(device) if device is not None else get_current_cuda_device()...)`

Reply:
I can't get it to cython compile with 

```
prefetch(self.c_obj.get()[0].data(),
                self.c_obj.get()[0].size(),
                dev,
                stream.view())
```

```
/home/coder/rmm/python/rmm/rmm/_lib/device_buffer.pyx:161:27: Cannot convert Python object to 'cuda_stream_view'
```

Reply:
It's complaining becuase in the function signature for this function `c_prefetch` the fourth argument is typed as a `Stream`, so when you call the underlying C function `prefetch` it knows the type of that object. Conversely, when you look at the Python `prefetch` implementation above, the `stream` object is not typed. You can fix that by doing a `cdef Stream stream =...` and it should work.

Reply:
Thanks!  Fixed.